### PR TITLE
[P5-A11-WP1] Create tests/fixtures/codex Package and All NDJSON Fixture Files

### PR DESCRIPTION
**Pull request by Trecek**
## Summary

Create a new `tests/fixtures/codex/` package containing five NDJSON fixture files representing Codex CLI session output scenarios (happy path, multi-turn, failure, reasoning, MCP tool calls). The package provides a `fixture_path()` helper and filename constants for downstream consumption by `tests/execution/backends/` test suites. A dedicated fixture integrity test file validates all acceptance criteria. The test-filter manifest is updated to route `.ndjson` file changes to the `execution/` test directory.

**Critical commit constraint:** All new files (NDJSON fixtures + `__init__.py` files + test file) and the manifest update MUST land in a single commit. The `test_manifest_pattern_matches_real_file` orphan check in `tests/infra/test_manifest_completeness.py` runs `git ls-files --cached` and will fail if the manifest pattern `tests/fixtures/codex/*.ndjson` is present but the NDJSON files are not yet tracked.

### Goal
Scaffold the fixture directory with constants, then create all five NDJSON fixtures.

### Context
- Phase: Codex CLI Backend Implementation (Milestone: 5-codex-cli-backend-implementation)
- Assignment: P5-A11 (Create Codex NDJSON test fixtures for happy-path, error, multi-turn, and edge cases)
- Depends on: None
- Depended on by: P5-A12-WP1

### Deliverables
- `tests/fixtures/__init__.py`
- `tests/fixtures/codex/__init__.py`
- `tests/fixtures/codex/happy_path_single_turn.ndjson`
- `tests/fixtures/codex/multi_turn_with_compaction.ndjson`
- `tests/fixtures/codex/turn_failed_error.ndjson`
- `tests/fixtures/codex/session_with_reasoning.ndjson`
- `tests/fixtures/codex/session_with_mcp_tool_call.ndjson`

### Acceptance Criteria
- CODEX_SCHEMA_VERSION == 1
- fixture_path returns correct Path
- All filename constants end in .ndjson
- Every line is valid JSON
- First line is thread.started with thread_id
- agent_message contains %%ORDER_UP%% as standalone line
- turn.completed has non-zero usage
- turn2 input_tokens > turn1 input_tokens (cumulative invariant)
- turn_failed has non-empty error.message
- reasoning fixture has reasoning_output_tokens>0
- mcp fixture has all 3 status transitions
- Both are valid NDJSON

### Architecture

```mermaid
%%{init: {'flowchart': {'nodeSpacing': 50, 'rankSpacing': 60, 'curve': 'basis'}}}%%
flowchart TB
    classDef cli fill:#1a237e,stroke:#7986cb,stroke-width:2px,color:#fff;
    classDef stateNode fill:#004d40,stroke:#4db6ac,stroke-width:2px,color:#fff;
    classDef handler fill:#e65100,stroke:#ffb74d,stroke-width:2px,color:#fff;
    classDef phase fill:#6a1b9a,stroke:#ba68c8,stroke-width:2px,color:#fff;
    classDef newComponent fill:#2e7d32,stroke:#81c784,stroke-width:2px,color:#fff;
    classDef output fill:#00695c,stroke:#4db6ac,stroke-width:2px,color:#fff;
    classDef detector fill:#b71c1c,stroke:#ef5350,stroke-width:2px,color:#fff;

    subgraph Fixtures ["tests/fixtures/codex/"]
        direction TB
        INIT["__init__.py<br/>━━━━━━━━━━<br/>CODEX_SCHEMA_VERSION=1<br/>5 filename constants<br/>fixture_path helper"]
        HP["happy_path_single_turn.ndjson<br/>━━━━━━━━━━<br/>Full lifecycle + ORDER_UP"]
        MT["multi_turn_with_compaction.ndjson<br/>━━━━━━━━━━<br/>2 turns + cumulative tokens"]
        TF["turn_failed_error.ndjson<br/>━━━━━━━━━━<br/>Partial item + turn.failed"]
        SR["session_with_reasoning.ndjson<br/>━━━━━━━━━━<br/>reasoning item + tokens>0"]
        MCP["session_with_mcp_tool_call.ndjson<br/>━━━━━━━━━━<br/>started/updated/completed"]
    end

    subgraph Tests ["tests/execution/backends/"]
        direction TB
        TFIX["test_codex_fixtures.py<br/>━━━━━━━━━━<br/>Fixture integrity validation"]
    end

    subgraph Manifest [".autoskillit/"]
        direction TB
        MANIF["test-filter-manifest.yaml<br/>━━━━━━━━━━<br/>+ tests/fixtures/codex/*.ndjson"]
    end

    subgraph Consumers ["Existing Consumers"]
        direction TB
        PARSER["test_codex_result_parser.py<br/>━━━━━━━━━━<br/>_scan_codex_ndjson tests"]
        BACKEND["test_codex_backend.py<br/>━━━━━━━━━━<br/>CodexStreamParser tests"]
    end

    INIT --> TFIX
    HP --> TFIX
    MT --> TFIX
    TF --> TFIX
    SR --> TFIX
    MCP --> TFIX
    MANIF -.->|routes changes to| Tests
    TFIX -.->|validates| Fixtures
    HP -.->|future: consumed by| PARSER
    HP -.->|future: consumed by| BACKEND

    class INIT,HP,MT,TF,SR,MCP newComponent;
    class TFIX newComponent;
    class MANIF phase;
    class PARSER,BACKEND handler;
```

**Color Legend:**
| Color | Category | Description |
|-------|----------|-------------|
| Green | New Component | New files created by this task |
| Purple | Modified | Existing files with additions |
| Orange | Consumer | Existing test files that will consume fixtures |

Closes #2509

## Implementation Plan

Plan file: `/home/talon/projects/autoskillit-runs/impl-20260521-075117-926197/.autoskillit/temp/make-plan/p5_a11_wp1_create_tests_fixtures_codex_plan_2026-05-21_081000.md`

🤖 Generated with [Claude Code](https://claude.com/claude-code) via AutoSkillit

## Token Usage Summary

| Step | Model | count | uncached | output | cache_read | peak_ctx | turns | cache_write | time |
|------|-------|-------|----------|--------|------------|----------|-------|-------------|------|
| plan | claude-sonnet-4-6 | 1 | 4.6k | 25.1k | 1.4M | 101.1k | 341 | 83.1k | 13m 35s |
| verify | claude-opus-4-6 | 1 | 37 | 6.9k | 864.3k | 58.6k | 73 | 43.4k | 4m 24s |
| implement* | MiniMax-M2.7-highspeed | 1 | 762.1k | 10.5k | 863.4k | 34.7k | 104 | 60.9k | 7m 14s |
| prepare_pr* | MiniMax-M2.7 | 1 | 45.3k | 4.0k | 207.9k | 0 | 17 | 0 | 1m 11s |
| compose_pr* | MiniMax-M2.7 | 1 | 126.0k | 3.5k | 407.7k | 26.7k | 31 | 2.9k | 2m 34s |
| review_pr | claude-sonnet-4-6 | 3 | 677 | 106.8k | 2.1M | 90.4k | 118 | 229.6k | 21m 50s |
| resolve_review | claude-sonnet-4-6 | 2 | 414 | 31.1k | 2.9M | 77.2k | 116 | 119.4k | 13m 36s |
| **Total** | | | 939.3k | 187.8k | 8.7M | 101.1k | | 539.4k | 1h 4m |

\* *Step used a non-Anthropic provider; caching behavior may differ.*

## Token Efficiency

| Step | LoC Changed | cache_read/LoC | cache_write/LoC | output/LoC |
|------|-------------|----------------|-----------------|------------|
| plan | 0 | — | — | — |
| verify | 0 | — | — | — |
| implement | 312 | 2767.4 | 195.1 | 33.6 |
| prepare_pr | 0 | — | — | — |
| compose_pr | 0 | — | — | — |
| review_pr | 0 | — | — | — |
| resolve_review | 28 | 102307.7 | 4265.0 | 1110.7 |
| **Total** | **340** | 25510.6 | 1586.4 | 552.4 |

## Model Usage Breakdown

| Model | steps | uncached | output | cache_read | cache_write | time |
|-------|-------|----------|--------|------------|-------------|------|
| claude-sonnet-4-6 | 3 | 5.7k | 163.1k | 6.3M | 432.2k | 49m 3s |
| claude-opus-4-6 | 1 | 37 | 6.9k | 864.3k | 43.4k | 4m 24s |
| MiniMax-M2.7-highspeed | 1 | 762.1k | 10.5k | 863.4k | 60.9k | 7m 14s |
| MiniMax-M2.7 | 2 | 171.4k | 7.4k | 615.7k | 2.9k | 3m 46s |

### CODE REVIEW

File: tests/execution/backends/test_codex_fixtures.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+from tests.fixtures.codex import (
+    CODEX_SCHEMA_VERSION,
+    HAPPY_PATH_SINGLE_TURN,
+    MULTI_TURN_WITH_COMPACTION,
+    SESSION_WITH_MCP_TOOL_CALL,
+    SESSION_WITH_REASONING,
+    TURN_FAILED_ERROR,
+    fixture_path,
+)
+from tests.fixtures.codex import (
```
Comment (Trecek) [inline tests/execution/backends/test_codex_fixtures.py:20]: [warning] cohesion: The `__all__` import is split into a second `from tests.fixtures.codex import` block (L20-L22) separate from the first (L11-L19). Both import from the same module; they should be merged into one import block for consistency with standard import grouping practice.

File: tests/execution/backends/test_codex_fixtures.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    __all__ as CODEX_ALL,
+)
+
+pytestmark = [pytest.mark.layer("execution"), pytest.mark.small]
+
+ALL_FIXTURE_NAMES = [
```
Comment (Trecek) [inline tests/execution/backends/test_codex_fixtures.py:26]: [warning] cohesion: `ALL_FIXTURE_NAMES` (L26-L32) lists 5 names but `test_all_exports_count` (L53-L54) asserts `len(CODEX_ALL) == 7`. The local constant and the `__all__` assertion are tracking different sets with no comment explaining the discrepancy (7 exports vs 5 fixture names), creating a confusing asymmetry.

File: tests/execution/backends/test_codex_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    def test_fixture_path_returns_existing_file(self) -> None:
+        for name in ALL_FIXTURE_NAMES:
+            assert fixture_path(name).is_file()
+
+    def test_all_exports_count(self) -> None:
+        assert len(CODEX_ALL) == 7
+
+
+class TestCodexFixtureValidity:
+    def test_every_line_is_valid_json(self) -> None:
```
Comment (Trecek) [inline tests/execution/backends/test_codex_fixtures.py:58]: [warning] cohesion: `test_every_line_is_valid_json` inlines its own NDJSON parsing loop instead of using the `_load_events` helper already defined at L35. This duplicates parsing logic and breaks the established pattern used by every other test method in the file.

**Review by Trecek — COMMENTED** [review]
AutoSkillit PR Review — Verdict: approved_with_comments

**Review by Trecek — COMMENTED** [review]
AutoSkillit review: warning-only findings detected. See inline comments — no blocking changes required.
